crates/bevy_input/src/input.rs

```diff
@@ -45,52 +46,74 @@ use bevy_ecs::schedule::State;
 ///[`DetectChangesMut::bypass_change_detection`]: bevy_ecs::change_detection::DetectChangesMut::bypass_change_detection
 #[derive(Debug, Clone, Resource, Reflect)]
 #[reflect(Default)]
-pub struct Input<T: Clone + Eq + Hash + Send + Sync + 'static> {
+pub struct Input<T: FromReflect + Clone + Eq + Hash + Send + Sync + 'static> {
     /// A collection of every button that is currently being pressed.
     pressed: HashSet<T>,
     /// A collection of every button that has just been pressed.
     just_pressed: HashSet<T>,
     /// A collection of every button that has just been released.
     just_released: HashSet<T>,
+    /// To map logical keys with their respecting physical keys for reliability.
+    dynamic_map_value: HashMap<T, T>,
```

Comment:
This is a *very* interesting change. I'm not sure I agree with the design: why would the physical and logical keys be the same?

Reply:
This change is led by the inability from winit to be sure a key_pressed(Character("a")) leads to a key_released(Character("a")), so #8745 believes character("a") is still pressed.

To understand the issue, I invite you to try the example keyboard_input.

I have to keep track on original keycode to update its cached keypress status.

Now I could have made that change only for Input<Key> (logical key), but as the data model is shared, that would have led to code  duplication, which I thought was not worth it.

Another option would be to report it to winit, to evaluate if we can have a character released without the modifiers, or an event for key_change: from a modifier (lowercase) to another (uppercase) 🤔